src/Front/CDN.php

```diff
@@ -0,0 +1,290 @@
+<?php
+declare(strict_types=1);
+
+namespace RocketCDN\Front;
+
+use RocketCDN\Options\Options;
+
+class CDN {
+	/**
+	 * Options instance
+	 *
+	 * @var Options
+	 */
+	private $options;
+
+	/**
+	 * Home URL host
+	 *
+	 * @var string
+	 */
+	private $home_host = '';
+
+	/**
+	 * Instantiates the class
+	 *
+	 * @param Options $options Options instance.
+	 */
+	public function __construct( Options $options ) {
+		$this->options = $options;
+	}
+
+	/**
+	 * Setup output buffering
+	 *
+	 * @since 1.0
+	 *
+	 * @return void
+	 */
+	public function start_buffering() {
+		if ( ! $this->should_rewrite() ) {
+			return;
+		}
+
+		ob_start( [ $this, 'end_buffering' ] );
+	}
+
+	/**
+	 * Processes the buffer content and return it
+	 *
+	 * @param string $html HTML content.
+	 *
+	 * @return string
+	 */
+	public function end_buffering( $html ): string {
+		$html = $this->rewrite( $html );
+		$html = $this->rewrite_srcset( $html );
+
+		return $html;
+	}
+
+	/**
+	 * Checks if we should rewrite the content
+	 *
+	 * @since 1.0
+	 *
+	 * @return bool
+	 */
+	private function should_rewrite(): bool {
+		if (
+			! isset( $_SERVER['REQUEST_METHOD'] )
+			||
+			'GET' !== $_SERVER['REQUEST_METHOD']
+		) {
+			return false;
+		}
+
+		if (
+			is_admin()
+			||
+			is_customize_preview()
+			||
+			is_embed()
+		) {
+			return false;
+		}
+
+		return true;
+	}
+
+	/**
+	 * Search & Replace URLs with the CDN URLs in the provided content
+	 *
+	 * @since 1.0
+	 *
+	 * @param string $html HTML content.
+	 * @return string
+	 */
+	private function rewrite( string $html ): string {
```

Comment:
The private methods here will be more testable if they can be extracted to a couple other smaller classes:
`Rewriter` class could expose `rewrite()` and `rewrite_srcset()`.
`PathHelper` class could expose the other url/path transformation logic methods.

(The CDN class would be basically a controller without any real logic to test.)

Reply:
Per discussion, we'll split these out later if we need to.